cdk/lib/csp/public-asset-directives.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  getCognitoIdenityPoolEndpoint,
+  getCognitoUserPoolEndpoint,
+  getEventsControlPlaneEndpoint,
+  getEventsDataPlaneEndpoint,
+  getSiteWiseControlPlaneEndpoint,
+  getSiteWiseDataPlaneEndpoint,
+} from '../endpoints/aws-endpoints';
+
+export function getConnectSrcDirective(
+  coreServiceUrl: string,
+  region: string,
+): string {
+  const directiveName = 'connect-src';
+
+  const sources = [
+    coreServiceUrl,
+    getCognitoIdenityPoolEndpoint(region),
+    getCognitoUserPoolEndpoint(region),
+    getEventsControlPlaneEndpoint(region),
+    getEventsDataPlaneEndpoint(region),
+    getSiteWiseControlPlaneEndpoint(region),
+    getSiteWiseDataPlaneEndpoint(region),
+  ];
+
+  const policy = [directiveName, ...sources].join(' ');
+
+  return `${policy};`;
+}
+
+export function getDefaultSrcDirective(): string {
+  return "default-src 'none';";
+}
+
+export function getFontSrcDirective(): string {
+  return "font-src 'self' data:;";
+}
+
+export function getImgSrcDirective(): string {
+  return "img-src 'self' data:;";
+}
+
+export function getScriptSrcDirective(): string {
+  // FIXME: 'unsafe-eval' source to be removed after removal of NPM dependency box-intersect
```

Comment:
Nit: Consider making this comment more specific (to IoT App Kit dashboard) and/or linking a GitHub issue.